### DISCUSSION TITLE
How to make external code by libtool

### DISCUSSION THREAD

**Discussion by lvkas521424** [idaholab/moose#27877] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello, everone :
Currently, as far as I can tell from my limited knowledge of moose, the code guidelines are given to the build.mk file, and you can use libtool directly to compile this code and generate external dynamic link libraries for mutiapp to call. I have successfully compiled Cpp and Gfortran code with it and completed its function calls. But now there is a problem that bothers me, currently I have a Fortran based program, but it can only be compiled by ifort, but when I switch the server to ifort environment, compiling it by using moose's libtool tool will show me an error, whereas when I compile it directly with ifort externally, I won't have any problem. I would like to ask you guys why this is the case.

My app is named program. The display of the error report looks like this :
![libtool编译ifort下Fortran代码报错](https://github.com/idaholab/moose/assets/166678573/ee049de0-af88-4c60-9de0-3c1c5da24ec3)

yours,
JiahuiLv

**Comment by GiudGiud** [comment, marked as answer]
Hello

We don't support the intel compilers. Pleasery to make your external program be compilable with gfortran and the rest of our compilers, so you can use a consistent set of compilers across the entire program.

Guillaume

**Comment by lvkas521424** [comment]
Thanks for your reply, I'm sorry, I just assumed before that moose's petsc module doesn't support ifort, I will modify my program to accommodate moose doing external coupling.

yours
Jiahui Lv
